code/modules/events/patient_zero.dm

```diff
@@ -0,0 +1,52 @@
+/datum/round_event_control/patient_zero
+	name = "Patient Zero"
+	typepath = /datum/round_event/ghost_role/patient_zero
+	weight = 5
+	max_occurrences = 1
+	min_players = 20
+	earliest_start = 50 MINUTES
+
+/datum/round_event/ghost_role/patient_zero
+	announceWhen	= 500
+	minimum_required = 1
+	role_name = "patient zero"
+	fakeable = TRUE
+
+/datum/round_event/ghost_role/patient_zero/announce(fake)
+	priority_announce("Unidentified lifesigns detected coming aboard [station_name()]. Secure any exterior access, including ducting and ventilation.", "Lifesign Alert", ANNOUNCER_ALIENS)
+
+/datum/round_event/ghost_role/patient_zero/spawn_role()
+	var/list/candidates = get_candidates(ROLE_ALIEN, null, ROLE_ALIEN)
+	if(!candidates.len)
+		return NOT_ENOUGH_PLAYERS
+
+	var/mob/dead/selected = pick(candidates)
+
+	var/list/spawn_locs = list()
+	for(var/obj/effect/landmark/carpspawn/C in GLOB.landmarks_list)
```

Comment:
If the patient zero spawns in a carp spawn, wouldn't they be just stuck floating in space? Can they spacewalk?

Reply:
They get spawned in space, and launched towards the station.